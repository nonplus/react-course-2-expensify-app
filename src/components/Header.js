import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { logoutAsync } from "../actions/auth";

export const Header = ({ logoutAsync }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/" exact={true}>
          <h1>Expensify</h1>
        </Link>
        <button className="button button--link" onClick={logoutAsync}>
          Logout
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  logoutAsync: () => dispatch(logoutAsync())
});

export default connect(undefined, mapDispatchToProps)(Header);
