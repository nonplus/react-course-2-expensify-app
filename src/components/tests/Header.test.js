import { shallow } from "enzyme";
import React from "react";
import { Header } from "../Header";

describe("<Header />", () => {
  let wrapper, logoutAsync;
  beforeEach(() => {
    logoutAsync = jest.fn();
    wrapper = shallow(<Header logoutAsync={logoutAsync} />);
  });

  it("should render Header correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("<button/>", () => {
    let button;
    beforeEach(() => (button = wrapper.find("button")));

    describe("onClick", () => {
      it("should call logoutAsync", () => {
        button.simulate("click");
        expect(logoutAsync).toHaveBeenCalled();
      });
    });
  });
});
