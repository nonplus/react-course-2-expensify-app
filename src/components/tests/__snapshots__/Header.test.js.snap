// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        exact={true}
        replace={false}
        to="/"
      >
        <h1>
          Expensify
        </h1>
      </Link>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
